### PR TITLE
Support pool ipam

### PR DESCRIPTION
**Pull request by yaocw2020**
dependency: https://github.com/harvester/load-balancer-harvester/pull/3

### CODE REVIEW

File: doc/load-balancer-request-parameters.md [hunk 1]
```diff
@@ -1,12 +1,10 @@
...
-### Load Balancer Type
-The Harvester load balancer has two types.
-- Internal: The load balancer will get an IPv4 address that can only be accessed inside Harvester.
-- External: The load balancer will get an IPv4 address from the DHCP server. Whether the IPv4 address is in the LAN or WAN depends on the DHCP server configuration.
-
-We can configure the type by the annotation key `cloudprovider.harvesterhci.io/type`. Its value can be `internal` and `external`.
+### IPAM
+We can configure the IPAM mode by the annotation key `loadbalancer.harvesterhci.io/ipam`. Its value can be `pool` and `DHCP`. Defaults to `pool`.
+- pool: Uses should configure a IPAM pool in the Harvester. The Harvester LoadBalancer will allocate an address for the service from the IP pool.
+- DHCP: It requires a DHCP server. The Harvester LoadBalancer will request an address for the service from the DHCP server.
```
Comment (gitlawr) [inline doc/load-balancer-request-parameters.md:7]:
```
IPAM pool
=>
IP address pool
```
```
The Harvester LoadBalancer will allocate an address for the service
=>
Harvester will allocate an address for the load balancer
```

File: deploy/harvester-cloud-provider/templates/deployment.yaml [hunk 2]
```diff
@@ -47,5 +47,6 @@ spec:
       {{- end }}
       volumes:
         - name: cloud-config
-          secret:
-            secretName: cloud-config
\ No newline at end of file
+          hostPath:
+            path: /root/cloud-config.yaml
+            type: FileOrCreate
```
Comment (gitlawr) [inline deploy/harvester-cloud-provider/templates/deployment.yaml:52]: How is the file generated?

File: pkg/cloud-controller-manager/loadBalancer.go [hunk 3]
```diff
@@ -138,6 +131,7 @@ func (l *LoadBalancerManager) EnsureLoadBalancer(ctx context.Context, clusterNam
 	case <-time.After(defaultWaitIPTimeout):
 		return nil, fmt.Errorf("wait ip timeout")
 	case ip := <-ipChan:
+		klog.Infof("ip: %s", ip)
```
Comment (gitlawr) [inline pkg/cloud-controller-manager/loadBalancer.go:134]: make this log more readable

File: deploy/harvester-cloud-provider/templates/deployment.yaml (hunk 2 above)
Comment (yaocw2020) [reply deploy/harvester-cloud-provider/templates/deployment.yaml:52]: We have not confirmed the location of the cloud-config. It depends on the RKE2. I just fill a path randomly. I will change it to a variable.

File: pkg/cloud-controller-manager/loadBalancer.go (hunk 3 above)
Comment (yaocw2020) [reply pkg/cloud-controller-manager/loadBalancer.go:134]: Sorry, to delete it.

File: doc/load-balancer-request-parameters.md [hunk 4]
```diff
@@ -1,12 +1,10 @@
...
 
-### Load Balancer Type
-The Harvester load balancer has two types.
-- Internal: The load balancer will get an IPv4 address that can only be accessed inside Harvester.
-- External: The load balancer will get an IPv4 address from the DHCP server. Whether the IPv4 address is in the LAN or WAN depends on the DHCP server configuration.
-
-We can configure the type by the annotation key `cloudprovider.harvesterhci.io/type`. Its value can be `internal` and `external`.
+### IPAM
+We can configure the IPAM mode by the annotation key `loadbalancer.harvesterhci.io/ipam`. Its value can be `pool` and `DHCP`. Defaults to `pool`.
+- pool: Uses should configure an IP address pool in the Harvester. The Harvester LoadBalancer will allocate an address from the IP address poll for the load balancer.
```
Comment (guangbochen) [inline doc/load-balancer-request-parameters.md:6]: \Uses\Users

File: deploy/harvester-cloud-provider/templates/deployment.yaml (hunk 2 above)
Comment (gitlawr) [reply deploy/harvester-cloud-provider/templates/deployment.yaml]: Does it mean this change is not working at the moment? Is there a related rke2 issue?

Comment (yaocw2020) [reply deploy/harvester-cloud-provider/templates/deployment.yaml]: https://github.com/rancher/rancher/issues/33611

**Review by guangbochen — APPROVED** [review]
LGTM

**Pull request merged**